MacOS build fails if configured without cocoa

On macOS, `-Dcocoa=Off` will cause the build to fail with
```
[ 77%] Linking CXX shared library ../lib/libCore.so
Undefined symbols for architecture arm64:
  "TMacOSXSystem::TMacOSXSystem()", referenced from:
      ROOT::new_TMacOSXSystem(void*) in G__Core.cxx.o
      ROOT::newArray_TMacOSXSystem(long, void*) in G__Core.cxx.o
  "TMacOSXSystem::~TMacOSXSystem()", referenced from:
      ROOT::newArray_TMacOSXSystem(long, void*) in G__Core.cxx.o
      ROOT::deleteArray_TMacOSXSystem(void*) in G__Core.cxx.o
  "typeinfo for TMacOSXSystem", referenced from:
      ROOT::GenerateInitInstanceLocal(TMacOSXSystem const*) in G__Core.cxx.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libCore.so] Error 1
make[1]: *** [core/CMakeFiles/Core.dir/all] Error 2
make: *** [all] Error 2
```

--- a/core/macosx/CMakeLists.txt
+++ b/core/macosx/CMakeLists.txt
@@ -8,7 +8,7 @@
 # CMakeLists.txt file for building ROOT core/macosx package
 ############################################################################
 
-if(NOT APPLE)
+if(NOT (APPLE AND cocoa))
   return()
 endif()
 
@@ -20,19 +20,16 @@ target_link_libraries(Core PRIVATE
   "-F/System/Library/PrivateFrameworks -framework CoreSymbolication"
 )
 
-if(cocoa)
-  target_link_libraries(Core PRIVATE "-framework Cocoa")
+target_link_libraries(Core PRIVATE "-framework Cocoa")
 
-  ROOT_OBJECT_LIBRARY(Macosx
-    src/CocoaUtils.mm
-    src/TMacOSXSystem.mm
-  )
+ROOT_OBJECT_LIBRARY(Macosx
+  src/CocoaUtils.mm
+  src/TMacOSXSystem.mm
+)
 
-  target_compile_options(Macosx PRIVATE -ObjC++)
-  target_include_directories(Macosx PRIVATE ../unix/inc)
+target_compile_options(Macosx PRIVATE -ObjC++)
+target_include_directories(Macosx PRIVATE ../unix/inc)
 
-  target_sources(Core PRIVATE $<TARGET_OBJECTS:Macosx>)
-
-endif()
+target_sources(Core PRIVATE $<TARGET_OBJECTS:Macosx>)
 
 ROOT_INSTALL_HEADERS()

